vim_menutrans_merger.py

```diff
@@ -0,0 +1,104 @@
+            elif len(new_word_list) > 1 and new_word_list[0].startswith("g:menutrans_"):
+                translated_dict[new_word_list[0].lower()] = (
+                        line_number, translation_file, new_word_list[0], line[match_index[0].end():])
+            elif (len(new_word_list) > 1 and new_word_list[0] == "let" and 
+                    new_word_list[1].startswith("g:menutrans_")):
+                translated_dict[new_word_list[1].lower()] = (
+                        line_number, translation_file, new_word_list[1], line[match_index[2].end():])
+
+def extract_translated_message(template_file, translation_file):
+    """Get the translated files' content and put it into translated_dict"""
+
+    translated_dict = {}
+    make_translated_dict(translation_file, translated_dict)
+    replace_template_translation(template_file, translated_dict)
+
+def replace_template_translation(template_file, translated_dict):
+    """Replace the translations in template_file with the content in translated_dict"""
+
+    newContent = ""
+    # Encode with "latin1", because we don't care about the correctness of 
+    # non-ASCII character.
+    with open(template_file, encoding='UTF-8') as f:
+        for line_number, line in enumerate(f):
+            line = line.strip()
+            if not line:
+                newContent += "\n"
+                continue
+
+            # Split a line by white spaces but not backslash-escaped spaces.
+            new_word_list = re.findall(r"(?:\\ |[^ \t])+", line)
+            match_index = [each for each in re.finditer(r"(?:\\ |[^ \t])+", line)]
+            hasTranslated = True
+            
+            if len(new_word_list) > 1 and new_word_list[0] in MENUTRANS_COMMANDS:
+                if (new_word_list[1].lower() in translated_dict):
+                    line = line[:match_index[1].end() + 1] + translated_dict[new_word_list[1].lower()][3]
+                else:
+                    hasTranslated = False
+            elif len(new_word_list) > 1 and new_word_list[0].startswith("g:menutrans_"):
+                if (new_word_list[0].lower() in translated_dict):
+                    line = line[:match_index[0].end() + 1] + translated_dict[new_word_list[0].lower()][3]
+                else:
+                    hasTranslated = False
+            elif (len(new_word_list) > 1 and new_word_list[0] == "let" and 
+                    new_word_list[1].startswith("g:menutrans_")):
+                if (new_word_list[1].lower() in translated_dict):
+                    line = line[:match_index[2].end() + 1] + translated_dict[new_word_list[1].lower()][3]
+                else:
+                    hasTranslated = False
+            
+            if not hasTranslated:
+                line = '" ' + line
+
+            newContent += str(line_number + 1) + line
+            newContent += "\n"
+
+    with open("test.vim", mode="w", encoding="UTF-8") as f:
```

Comment:
Always write to `test.vim`? We need to overwrite the original translation file with the merged result.